extensions/common/vault/vault-hashicorp/src/test/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClientTest.java

```diff
@@ -9,228 +9,447 @@
     private static final String HEALTH_PATH = "sys/health";
-    private static final Duration TIMEOUT = Duration.ofSeconds(30);
+    private static final String VAULT_TOKEN = UUID.randomUUID().toString();
+    private static final long VAULT_TOKEN_TTL = 5L;
+    private static final long RENEW_BUFFER = 4L;
+    private static final String CUSTOM_SECRET_PATH = "v1/test/secret";
+    private static final String KEY = "key";
+    private static final String DATA_KEY = "data";
+    private static final String RENEWABLE_KEY = "renewable";
+    private static final String AUTH_KEY = "auth";
+    private static final String LEASE_DURATION_KEY = "lease_duration";
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
-
-    private final EdcHttpClient edcClientMock = mock(EdcHttpClient.class);
-
-    @BeforeEach
-    void setup() {
-
+    private static final HashicorpVaultConfigValues HASHICORP_VAULT_CLIENT_CONFIG_VALUES = HashicorpVaultConfigValues.Builder.newInstance()
+            .url(VAULT_URL)
+            .healthCheckPath(HEALTH_PATH)
+            .healthStandbyOk(false)
+            .token(VAULT_TOKEN)
+            .ttl(VAULT_TOKEN_TTL)
+            .renewBuffer(RENEW_BUFFER)
+            .secretPath(CUSTOM_SECRET_PATH)
+            .build();
+
+    private final EdcHttpClient httpClient = mock();
+    private final Monitor monitor = mock();
+    private final HashicorpVaultClient vaultClient = new HashicorpVaultClient(
+            httpClient,
+            OBJECT_MAPPER,
+            monitor,
+            HASHICORP_VAULT_CLIENT_CONFIG_VALUES);
+
+    @Nested
+    class HealthCheck {
+        @Test
+        void doHealthCheck_whenHealthCheckReturns200_shouldSucceed() throws IOException {
+            // prepare
+            var body = """
+                    {
+                    "initialized": true,
+                    "sealed": false,
+                    "standby": false,
+                    "performance_standby": false,
+                    "replication_performance_mode": "mode",
+                    "replication_dr_mode": "mode",
+                    "server_time_utc": 100,
+                    "version": "1.0.0",
+                    "cluster_name": "name",
+                    "cluster_id": "id"
+                    }
+                    """;
+            var response = new Response.Builder()
+                    .code(200)
+                    .message("any")
+                    .body(ResponseBody.create(body, MediaType.get("application/json")))
+                    .protocol(Protocol.HTTP_1_1)
+                    .request(new Request.Builder().url("http://any").build())
+                    .build();
+            when(httpClient.execute(any(Request.class))).thenReturn(response);
+
+            // invoke
+            var healthCheckResponseResult = vaultClient.doHealthCheck();
+
+            // verify
+            assertThat(healthCheckResponseResult.succeeded()).isTrue();
```

Comment:
please consider using the `AbstractResultAssert`, which allows to directly assert on a result:
```java
assertThat(healthCheckResponseResult).isSucceeded();
// or:
assertThat(someResult).isFailed().detail().isEqualTo("foobar");
```